source/exe/BUILD

```diff
@@ -48,7 +48,10 @@ envoy_cc_library(
 
 envoy_cc_library(
     name = "envoy_main_entry_lib",
-    srcs = ["main.cc"],
+    srcs = select({
+        "//bazel:windows_x86_64": ["win32/main.cc"],
```

Comment:
Can we do an `envoy_platform_deps` here and elsewhere? You might need to change the structure to have this as an explicit library.